File: 3/puz1.cpp
// Project: puz1
#include <bits/stdc++.h>
using namespace std;

vector<string> lines;

int checkSlope(int dx, int dy) {
	int C = 0;

	int cx=0;
	int cy=0;
	int wid = 31;

	for(string line : lines) {
		if(!(cy % dy)) {
			if(line[cx] == '#'){
				C++;
			}
			cx +=dx;
			cx %=wid;
		}
		cy ++;
	}

	return C;
}


int main() {
	ifstream inp("in1");
	while(inp.peek() != EOF) {
		string line;
		getline(inp,line);
		lines.push_back(line);
	}
	inp.close();

	long T = 1;
	T*= checkSlope(1,1);
	T*= checkSlope(3,1);
	T*= checkSlope(5,1);
	T*= checkSlope(7,1);
	T*= checkSlope(1,2);

	cout << T << '\n';
	cout << checkSlope(1,1) << '\n';
	cout << checkSlope(3,1) << '\n';
	cout << checkSlope(5,1) << '\n';
	cout << checkSlope(7,1) << '\n';
	cout << checkSlope(1,2) << '\n';
}

File: 12/puz1.py
inp = open("./in1")
lines = inp.readlines()
inp.close()

px = 0
py = 0

cx = 10
cy = 1

angle = 0

for line in lines:
    if line[0] == 'F':
        num = int(line[1:])
        for i in range(num):
            px += cx
            py += cy
        # if angle == 0:
        #     cx += num
        # elif angle == 90:
        #     cy -= num
        # elif angle == 180:
        #     cx -= num
        # elif angle == 270:
        #     cy += num
    if line[0] == 'L':
        num = int(line[1:])
        while num > 0:
            tmp =cx
            cx = -cy
            cy = tmp
            num-=90

        # angle = (angle - num + 360) % 360
    if line[0] == 'R':
        num = int(line[1:])
        while num > 0:
            tmp =cx
            cx = cy
            cy = -tmp
            num-=90
        # angle = (angle + num + 360) % 360
    if line[0] == 'N':
        num = int(line[1:])
        cy+=num
    if line[0] == 'S':
        num = int(line[1:])
        cy-=num
    if line[0] == 'E':
        num = int(line[1:])
        cx+=num
    if line[0] == 'W':
        num = int(line[1:])
        cx-=num

print(str(px) + " " + str(py))
print(abs(px) + abs(py))

